content/guides/code-splitting-import.md

```diff
@@ -144,6 +144,28 @@ module.exports = {
 };
 ```
 
+## `import` imports the entire module namespace
+
+Consider the following two examples:
+
+```js
+// Example 1: top-level import
+import * as Component from './component';
+// Example 2: Code-splitting with import()
+import('./component').then(Component => /* ... */);
+```
+
+`Component` in both of those cases resolves to the same thing, meaning in the case of using `import()` with ES2015 modules you have to explicity access default and named exports:
+
+```js
+async function main() {
+  // Destructuring example
+  const { default: Component } = await import('./component');
```

Comment:
> Have you noted the `async`/`await` keywords?

I did see the `async`/`await`. But I also saw the line below it, which won't run until this promise resolves and the desctructuring assignment occurs, resulting in a synchronyous operation. Perhaps it would be better to split the two examples into separate methods or find a way to run them all in parallel for better performance.